lib/evm_database_ops.rb

```diff
@@ -119,15 +118,19 @@ def self.restore(db_opts, connect_opts = {})
         backup_folder = action == :dump ? "db_dump" : "db_backup"
         uri = File.join(connect_opts[:uri], backup_folder, connect_opts[:remote_file_name])
       end
+    else
+      uri = db_opts[:local_file]
+    end
 
-      session = MiqGenericMountSession.new_session(connect_opts)
-      db_opts[:local_file] = session.uri_to_local_path(uri)
+    MiqFileStorage.with_interface_class(connect_opts) do |file_storage|
+      send_args = [uri, db_opts[:byte_count]].compact
+      file_storage.send(STORAGE_ACTIONS_TO_METHODS[action], *send_args) do |input_path|
```

Comment:
`input_path` here is the IO stream that the actual backup command will either write to or read from, correct?

Reply:
Yup.

Reply:
Yeah, that seems reasonable.  I will add some docs.